Libraries/LibC/netdb.cpp

```diff
@@ -219,13 +276,77 @@ struct servent* getservbyport(int port, const char* protocol)
 }
 void setservent(int stay_open)
 {
-    dbg() << "FIXME: getservbyname(\"" << stay_open << "\")";
-    ASSERT_NOT_REACHED();
+    if(!services_file) {
+        services_file = fopen(serv_file_loc, "r");
+
+        if(!services_file) {
+            perror("error opening services file");
+            return;
+        }
+    }
+    rewind(services_file);
+    keep_service_file_open = stay_open;
 }
 void endservent()
 {
-    dbg() << "FIXME: getservbyname()";
-    ASSERT_NOT_REACHED();
+    if(!services_file) {
+        return;
+    }
+    fclose(services_file);
+    services_file = nullptr;
+    keep_service_file_open = 0;
+}
+
+//Fill the service entry buffer with the information contained in the currently read line, returns true if successfull, false if failure occurs.
+static bool fill_getserv_buffer(char* line, ssize_t read)
+{
+    //Splitting the line by tab delimiter and filling the servent buffers name, port, and protocol members.
+    auto split_line = String(line, read).split('\t');
+    strncpy(__getserv_name_buffer, split_line[0].characters(), split_line[0].length());
```

Comment:
`strncpy` with a `String::length()` will not include a null terminator.
Also this way of using `strncpy()` seems kinda pointless -- we're not verifying that `split_line[0].length()` fits in `sizeof(__getserv_name_buffer)`.

Also, if `String::split()` returns an empty vector, index `0` will not be valid.